src/pkgcheck/checks/codingstyle.py

```diff
@@ -79,6 +79,64 @@ def feed(self, pkg):
                     yield DeprecatedEapiCommand(name, line=call, lineno=lineno+1, eapi=pkg.eapi, pkg=pkg)
 
 
+class EbeginMissingEend(results.LineResult, results.Style):
+    """Ebuild calls ebegin in a phase, but is missing a call to eend"""
+
+    def __init__(self, func_name, **kwargs):
+        super().__init__(**kwargs)
+        self.func_name = func_name
+
+    @property
+    def desc(self):
+        return f'Call to ebegin with a missing call to eend ({self.func_name}), on line {self.lineno}'
+
+
+class EendMissingEbegin(results.LineResult, results.Style):
+    """Ebuild calls eend in a phase, but is missing a call to ebegin"""
+
+    def __init__(self, func_name, **kwargs):
+        super().__init__(**kwargs)
+        self.func_name = func_name
+
+    @property
+    def desc(self):
+        return f'Call to eend, but missing preceding call to ebegin ({self.func_name}), on line {self.lineno}'
+
+
+class EbeginEendCheck(Check):
+    """Scan ebuild for ebegin with missing eend, and vice versa"""
+    #probably also should try to see if you can handle scoping at all, or at least use a stack design to handle multiple calls
+
+    _source = sources.EbuildParseRepoSource
+    known_results = frozenset([EbeginMissingEend, EendMissingEbegin])
```

Comment:
I'd combine both of these results into one (e.g. UnmatchedEbeginEend) using a stack design or similar to find them in the check like I mentioned before.

Reply:
I'm working on it, going for the stack design, but it's proving a bit of a challenge for me :-)